simulation/world.py

```diff
@@ -249,8 +281,11 @@ def __init__(self, source, target, message: str) -> None:
 class NoiseEvent:
     """Encapsulates a single noise event"""
 
-    def __init__(self, location: Position) -> None:
-        self._location = location
+    def __init__(self, location: Position, source=None, time=0) -> None:
+        self.time = time
+        self.location = location
+        self.source = source
+        self.drawn = False
```

Comment:
The `self.drawn` probably shouldn't be part of the simulation, but if we fix the issue above it shouldn't be necessary anyway.

Reply:
I removed it in the later commit because that approach wasnt nice